satpy/_compat.py

```diff
@@ -88,3 +88,9 @@ def __get__(self, instance, owner=None):  # noqa
     # numpy <1.20
     from numpy import dtype as DTypeLike  # noqa
     from numpy import ndarray as ArrayLike  # noqa
+
+
+try:
+    from functools import cache  # type: ignore
+except ImportError:
+    from functools import lru_cache as cache  # noqa
```

Comment:
Maybe these could be removed, they are a leftover from my cache testings I think.

Reply:
I'm actually using the `self.method = cache(self._method)` trick around here https://github.com/pytroll/satpy/blob/d73bdde89d5ba21a5ef9ceaf75aa97650e47783c/satpy/readers/yaml_reader.py#L1383, see also https://github.com/pytroll/satpy/pull/2287/commits/132f9a32e21323b4dcef8c399c25cdbfdd6428c3, so we should keep it.

Reply:
Oooh, yes! I missed that.